core/tests/build_sources/extensions/base.py

```diff
@@ -70,75 +109,75 @@ class BaseInteraction:
     # TODO(chiangs): remove once classifier_services is generalized.
-    is_trainable = False
+    is_trainable: bool = False
     # Additional JS library dependencies that should be loaded in pages
     # containing this interaction. These should correspond to names of files in
     # feconf.DEPENDENCIES_TEMPLATES_DIR. Overridden in subclasses.
-    _dependency_ids = []
+    _dependency_ids: List[str] = []
     # The type of answer (as a string) accepted by this interaction, e.g.
     # 'CodeEvaluation'. This should be None for linear and terminal
     # interactions.
-    answer_type = None
+    answer_type: Optional[str] = None
     # Customization arg specifications for the component, including their
     # descriptions, schemas and default values. Overridden in subclasses.
-    _customization_arg_specs = []
+    _customization_arg_specs: List[CustomizationArgSpecsDict] = []
     # Specs for desired visualizations of recorded state answers. Overridden
     # in subclasses.
-    _answer_visualization_specs = []
+    _answer_visualization_specs: List[Dict[str, str]] = []
     # Instructions for using this interaction, to be shown to the learner. Only
     # relevant for supplemental interactions.
-    instructions = None
+    instructions: Optional[str] = None
     # Instructions for using this interaction, to be shown to the learner. Only
     # shows up when view port is narrow. Only relevent for supplemental
     # interactions.
-    narrow_instructions = None
+    narrow_instructions: Optional[str] = None
     # Whether the answer is long, and would benefit from being summarized.
-    needs_summary = False
+    needs_summary: bool = False
     # The heading for the 'default outcome' section in the editor. This should
     # be None unless the interaction is linear and non-terminal.
-    default_outcome_heading = None
+    default_outcome_heading: Optional[str] = None
     # Whether the solution feature supports this interaction.
-    can_have_solution = None
+    can_have_solution: bool = False
     # Whether to show a Submit button in the progress navigation area. This is
     # a generic submit button so do not use this if special interaction-specific
     # behavior is required. The interaction JS must also handle the
     # EVENT_PROGRESS_NAV_SUBMITTED event broadcast by this Submit button.
-    show_generic_submit_button = False
+    show_generic_submit_button: bool = False
 
     # Temporary cache for the rule definitions.
-    _cached_rules_dict = None
+    _cached_rules_dict: Optional[Dict[str, Dict[str, str]]] = None
 
     @property
-    def id(self):
+    def id(self) -> str:
         return self.__class__.__name__
 
     @property
-    def customization_arg_specs(self):
+    def customization_arg_specs(self) -> List[domain.CustomizationArgSpec]:
         return [
-            domain.CustomizationArgSpec(**cas)
+            domain.CustomizationArgSpec(**cas)  # type: ignore[no-untyped-call]
```

Comment:
Why is there an ignore here?

Reply:
Before this file was expected to be covered with type checking but after some discussion, we are now not covering these files because these files are just for testing.

A small discussion regarding this can be found here https://github.com/oppia/oppia/pull/15864#discussion_r942334850